routes/location.js

```diff
@@ -329,6 +337,91 @@ function getGCPCloudMetadata(callback) {
     req.end();
 }
 
+function getK8sCloudMetadata(callback) {
+    console.log('getK8sCloudMetadata');
+    // Set options to retrieve k8s api information
+    var node_name = process.env.MY_NODE_NAME;
+    console.log('Querying ' + node_name + ' for cloud data');
+
+    try {
+        var sa_token = fs.readFileSync('/var/run/secrets/kubernetes.io/serviceaccount/token');
+    } catch (err) {
+        console.log(err)
+    }
+
+    var headers = {
+        'Authorization': `Bearer ${sa_token}`
+    };
+
+    var genericOptions = {
+        host: 'kubernetes.default.svc',
+        port: 443,
+        path: `/api/v1/nodes/${node_name}`,
+        timeout: 10000,
+        ca: fs.readFileSync('/var/run/secrets/kubernetes.io/serviceaccount/ca.crt'),
+        headers: headers,
+    };
+
+    var cloudName = 'unknown',
+        zone = 'unknown';
+
+    var req = https.request(genericOptions, (zoneRes) => {
+        let error;
+
+        if (zoneRes.statusCode !== 200) {
+            error = new Error(`Request Failed.\n` +
+                `Status Code: ${zoneRes.statusCode}`);
+        }
+
+        if (error) {
+            console.log(error.message);
+            // consume response data to free up memory
+            zoneRes.resume();
+            callback(error, cloudName, zone);
+            return;
+        }
+
+        console.log(`STATUS: ${zoneRes.statusCode}`);
+        console.log(`HEADERS: ${JSON.stringify(zoneRes.headers)}`);
+        zoneRes.setEncoding('utf8');
+
+        var body = [];
+
+        zoneRes.on('data', (chunk) => {
+            body.push(chunk);
+        });
+        zoneRes.on('end', () => {
+            var metaData = JSON.parse(body.join(''));
+            console.log(`RESULT: ${metaData}`);
+            console.log('No more data in response.');
+            // Split ProviderID
+            var provider = metaData.spec.providerID;
+            cloudName = provider.split(":", 1); // Request was successful
```

Comment:
If this is not successful, we should return `unknown` for `cloudName` and `zone`.